mscore/scoreview.cpp

```diff
@@ -5540,14 +5546,96 @@ void ScoreView::cmdAddChordName()
       harmony->setTrack(cr->track());
       harmony->setParent(cr->segment());
       _score->undoAddElement(harmony);
-
       _score->select(harmony, SelectType::SINGLE, 0);
       startEdit(harmony);
       _score->setLayoutAll();
       _score->update();
       }
 
 //---------------------------------------------------------
+//   cmdAddAnnotation
+//---------------------------------------------------------
+
+void ScoreView::cmdAddAnnotation()
+      {
+      if (!_score->checkHasMeasures())
+            return;
+      if (noteEntryMode())          // force out of entry mode
+            sm->postEvent(new CommandEvent("note-input"));
+
+      _score->startCmd();
+        ChordRest* cr = _score->getSelectedChordRest();
+      if (!cr)
+            return;
+
+      TextAnnotation* t = new TextAnnotation(_score);
+      t->setTrack(cr->track());
+      t->setTextStyleType(TextStyleType::ANNOTATION);
+      t->setParent(cr->segment());
+      t->setColor(Qt::red);
+      if (t) {
+            _score->undoAddElement(t);
+            _score->select(t, SelectType::SINGLE, 0);
+            _score->endCmd();
+            startEdit(t);
+            }
+      else
+            _score->endCmd();
+
+      }
+
+//---------------------------------------------------------
+//   cmdAddRangeAnnotation
+//---------------------------------------------------------
+
+void ScoreView::cmdAddRangeAnnotation()
+      {
+      if (!_score->selection().isRange()) {
```

Comment:
seems all we really need to do differently is set stcik/etick/sstaff/estaff; the rest of the code is common to both case isn't it?  or could be made so?